README.md

```diff
@@ -1,138 +1,221 @@
+
+The value of `generate` can also be `none` as shown below.
 
 ```yaml
-# build_cuda-11.5_arch-x86_64.yaml
-channels:
-  - rapidsai
-  - nvidia
-dependencies:
-  - cudatoolkit==11.5.0
-  - django=3
-  - gcc_linux-64=9.*
-  - nvcc_linux-64=11.5
-  - pytorch
-  - sysroot_linux-64==2.17
-  - pip:
-      - random_dep
-      - snakes
-name: build_cuda-11.5_arch-x86_64
+files:
+  test:
+    generate: none
+    includes:
+      - test
 ```
 
-and
+When `generate: none` is used, the `conda_dir`, `requirements_dir` and `matrix` keys can be ommitted. The use case for `generate: none` is described in the [_Additional CLI Notes_](#additional-cli-notes) section below.
+
+### `channels` Key
+
+The top-level `conda` key specifies the channels that should be included in any generated `conda` environment files.
+
+It might look like this:
 
 ```yaml
-# build_cuda-11.6_arch-x86_64.yaml
 channels:
   - rapidsai
-  - nvidia
+  - conda-forge
+```
+
+In the absence of a `channels` key, some sensible defaults for RAPIDS will be used (see [constants.py](./src/rapids_dep_file_generator/constants.py)).
+
+### `dependencies` Key
+
+The top-level `dependencies` key is where the bifurcated dependency lists should be specified. Directly beneath the `dependencies` key are 3 unique keys:
+
+- `conda_requirements` - contains dependency lists that are the sames for both `conda` environment files and `requirements.txt` files
+- `conda` - contains dependency lists that are specific to `conda` environment files
+- `requirements` - contains dependency lists that are specific to `requirements.txt` files
+
+Each of the above keys has the following children keys:
+
+- `common` - contains dependency lists that are the same across CUDA versions and architectures
+- `<arch>-<cuda_version>` (i.e. `x86_64-11.5`) - contains dependency lists that are specific to the respective architecture and CUDA versions
+
+Below these keys are any number of arbitrarily named dependency lists (i.e. `build`, `test`, `libcuml_build`, `cuml_build`, etc.).
+
+An example of the above structure is exemplified below:
+
+```yaml
 dependencies:
-  - cudatoolkit==11.6.0
-  - django=3
-  - gcc_linux-64=9.*
-  - nvcc_linux-64=11.6
-  - pytorch
-  - sysroot_linux-64==2.17
-  - pip:
-      - random_dep
-      - snakes
-name: build_cuda-11.6_arch-x86_64
+  conda_requirements: # common dependencies between `conda` environment & requirements.txt files
+    common: # common between archs/cudas
+      build: # arbitrarily named dependency list
+        - common_build_dep
+      test: # arbitrarily named dependency list
+        - pytest
+    x86_64-11.5: # common dependencies specific to x86_64-11.5
+      build:
+        - a_random_x86_115_specific_dep
+  conda: # dependencies specific to conda environment files
+    common:
+      build:
+        - cupy
+        - pip: # supports `pip` key for `conda` environment files
```

Comment:
We shouldn't allow the `pip` keyword for `conda_requirements`, right?

Reply:
This is to account for entries like this:

https://github.com/rapidsai/cudf/blob/a67b718b220b7f722ff5f7a375618662419e5695/conda/environments/cudf_dev_cuda11.5.yml#L81-L83